src/test/java/com/google/cloud/flink/bigquery/FlinkBQConfigTest.java

```diff
@@ -0,0 +1,225 @@
+    ConfigOption<String> partition_require_filter =
+        ConfigOptions.key("partitionRequireFilter").stringType().defaultValue("");
+    ConfigOption<String> flink_version =
+        ConfigOptions.key("flinkVersion").stringType().defaultValue("1.11");
+    ConfigOption<Integer> max_parallelism =
+        ConfigOptions.key("maxParallelism").intType().defaultValue(1);
+    ConfigOption<String> selected_fields =
+        ConfigOptions.key("selectedFields").stringType().noDefaultValue();
+    ConfigOption<String> materialization_project =
+        ConfigOptions.key("materializationProject").stringType().noDefaultValue();
+    ConfigOption<String> materialization_dataset =
+        ConfigOptions.key("materializationDataset").stringType().noDefaultValue();
+    options.set(table, "bigquery-public-data.samples.shakespeare");
+    options.set(query, "select word,word_count from table");
+    options.set(filter, "word_count>100");
+    options.set(partition_field, "a");
+    options.set(partition_type, "DAY");
+    options.set(partition_expiration_ms, "999");
+    options.set(partition_require_filter, "true");
+    options.set(flink_version, "1.11.0");
+    options.set(max_parallelism, 99);
+    options.set(selected_fields, "word,word_count");
+    options.set(materialization_project, "vmp");
+    options.set(materialization_dataset, "vmd");
+
+    BigQueryDynamicTableFactory factory = new BigQueryDynamicTableFactory();
+    FlinkBigQueryConfig config =
+        FlinkBigQueryConfig.from(
+            factory.requiredOptions(),
+            factory.optionalOptions(),
+            (ReadableConfig) options,
+            defaultOptions,
+            hadoopConfiguration,
+            DEFAULT_PARALLELISM,
+            new org.apache.flink.configuration.Configuration(),
+            FLINK_VERSION,
+            Optional.empty());
+
+    assertThat(config.getTableId())
+        .isEqualTo(TableId.of("bigquery-public-data", "samples", "shakespeare"));
+    assertThat(config.getFilter()).isEqualTo(Optional.ofNullable("word_count>100"));
+    assertThat(config.getMaxParallelism()).isEqualTo(OptionalInt.of(99));
+    assertThat(config.getQuery()).isEqualTo(Optional.empty());
+    assertThat(config.getPartitionField()).isEqualTo(Optional.ofNullable("a"));
+    assertThat(config.getPartitionExpirationMs()).isEqualTo(OptionalLong.of(999));
+    assertThat(config.getPartitionRequireFilter()).isEqualTo(Optional.of(true));
+    assertThat(config.getDefaultParallelism()).isEqualTo(10);
+    assertThat(config.getMaterializationDataset()).isEqualTo(Optional.ofNullable("vmd"));
+    assertThat(config.getMaterializationExpirationTimeInMinutes()).isEqualTo(1440);
+    assertThat(config.getMaterializationProject()).isEqualTo(Optional.ofNullable("vmp"));
+    assertThat(config.getSelectedFields()).isEqualTo("word,word_count");
+  }
+
+  @Ignore
+  @Test
+  public void testInvalidCompressionCodec() {
```

Comment:
Why do we have this ignored test here? Let us please remove it

Reply:
Removed